olp-cpp-sdk-dataservice-read/test/unit/src/CatalogClientTest.cpp

```diff
@@ -754,6 +754,7 @@ class CatalogClientMockTest : public CatalogClientTestBase {
   void TearDown() {
     client_.reset();
     settings_.reset();
+    Mock::VerifyAndClearExpectations(network_mock_.get());
```

Comment:
If it is really important to check this resp to fail if you have unmet expectations then i would consider using StrictMock around NetworkMock. This way everything fails that is not expected or expected and not met.